### PR TITLE
[DB-158,211,266][risk=low] Database generation script updates 

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
1. Query update to fetch the rolled up counts and the path for snomed conditions from criteria.
2. Query addition to fetch unit specific counts for each biological sex from measurements.
3. Query update to fetch domain counts considering non-ppi and physical measurement concepts.

### CODE REVIEW

File: api/db-cdr/generate-cdr/make-bq-data.sh [hunk 1]
```diff
@@ -456,6 +456,18 @@ else
     exit 1
 fi
 
+##########################
+# Update rolled up counts #
+##########################
+echo "Updating rolled up counts in achilles results from criteria (for snomed conditions)"
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"UPDATE \`$OUTPUT_PROJECT.$OUTPUT_DATASET.achilles_results\` ar
+SET ar.count_value = cr1.est_count
```
Comment (calbach) [inline api/db-cdr/generate-cdr/make-bq-data.sh]: Is this overwriting a previous value in the table? What value was already there?

File: api/db-cdr/generate-cdr/make-bq-data.sh [hunk 2]
```diff
@@ -606,3 +616,23 @@ join \`$OUTPUT_PROJECT.$OUTPUT_DATASET.domain\` d2
 on d2.domain_id = c.domain_id
 and (c.count_value > 0 or c.source_count_value > 0)
 group by d2.domain_id,c.vocabulary_id"
+
+###############################################################
+# Update the path of concepts in achilles_results to fetch tree
```
Comment (calbach) [inline api/db-cdr/generate-cdr/make-bq-data.sh]: Could you expand this? I don't follow what fetch tree means.

File: api/db-cdr/generate-cdr/run-measurement-queries.sh [hunk 3]
```diff
@@ -994,6 +994,67 @@ and m1.measurement_source_concept_id > 0 and
 m1.measurement_source_concept_id not in (select distinct measurement_concept_id from \`${BQ_PROJECT}.${BQ_DATASET}.measurement\`)
 group by stratum_1,stratum_3,stratum_4,stratum_5"
 
+# Generating biological sex counts for measurement concepts for each unit
```
Comment (calbach) [inline api/db-cdr/generate-cdr/run-measurement-queries.sh]: I tried to stare at this for a while but I'm not really able to review it. Could you please expand this comment? I'm not really even sure what "unit" means in this context. Is this saying group by sex, then group by different measurement unit concept IDs, then count?

In the future, we should probably consider how we can try and refactor this file so it's more maintainable.

File: public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java [hunk 4]
```diff
@@ -619,10 +620,20 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
                         }else {
                                 unitSeperateAnalysis.add(aa);
                         }
-
                     }
                     isMeasurement = true;
                     conceptAnalysis.setMeasurementValueAgeAnalysis(unitSeperateAnalysis.stream().map(TO_CLIENT_ANALYSIS).collect(Collectors.toList()));
+                }else if(analysisId == MEASUREMENT_GENDER_UNIT_ANALYSIS_ID){
+                    HashMap<String,List<AchillesResult>> results = seperateUnitResults(aa);
```
Comment (calbach) [inline public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java]: nit: Better to use the more generic `Map` on the lhs, e.g. like you did with `List` below. HashMap is an irrelevant implementation detail.

File: public-ui/src/app/data-browser/concept-charts/concept-charts.component.ts [hunk 5]
```diff
@@ -41,7 +41,8 @@ export class ConceptChartsComponent implements OnChanges, OnInit, OnDestroy {
   selectedUnit: string;
   genderResults: AchillesResult[] = [];
   displayMeasurementGraphs = false;
-  toDisplayMeasurementGenderAnalysis: Analysis;
+  toDisplayMeasurementGenderAnalysis: any;
```
Comment (calbach) [inline public-ui/src/app/data-browser/concept-charts/concept-charts.component.ts]: Analysis doesn't work here anymore?

File: api/db-cdr/generate-cdr/make-bq-data.sh (hunk 1 above)
Comment (SrushtiGangireddy) [reply api/db-cdr/generate-cdr/make-bq-data.sh]: Previously there was the count which is the count of concept by itself and not the rolled up count.

File: api/db-cdr/generate-cdr/make-bq-data.sh (hunk 2 above)
Comment (SrushtiGangireddy) [reply api/db-cdr/generate-cdr/make-bq-data.sh]: I was fetching the path of concepts Eg: essential hypertension which can have path clinical finiding -> Finding by site -> Disorder of cardiovascular system -> Hypertensive disorder -> Essential hypertension. Storing the concept id path of tree in here.

File: api/db-cdr/generate-cdr/run-measurement-queries.sh (hunk 3 above)
Comment (SrushtiGangireddy) [reply api/db-cdr/generate-cdr/run-measurement-queries.sh]: yeah, Did not know of other way to it. Just building the queries based on previous ones. 
Measurement has different units (beats/ min, / sec and such) right, this query gets the biological sex count of each measurement for each unit. Eg: Heart rate can have 34000 male participants in which only 4000 could have their heart rate recorded in unit '/sec'.
This ftches the indvidual counts.

File: public-ui/src/app/data-browser/concept-charts/concept-charts.component.ts (hunk 5 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/concept-charts/concept-charts.component.ts]: Reverted it back.

File: api/db-cdr/generate-cdr/make-bq-data.sh (hunk 1 above)
Comment (calbach) [reply api/db-cdr/generate-cdr/make-bq-data.sh]: Please add this comment to the code

File: api/db-cdr/generate-cdr/make-bq-data.sh (hunk 2 above)
Comment (calbach) [reply api/db-cdr/generate-cdr/make-bq-data.sh]: Please add this comment to the code

File: api/db-cdr/generate-cdr/run-measurement-queries.sh (hunk 3 above)
Comment (calbach) [reply api/db-cdr/generate-cdr/run-measurement-queries.sh]: Please add this comment to the code

**Comment by calbach** [comment]
(snyk should be fixed after you rebase or merge master your branch)

**Comment by SrushtiGangireddy** [comment]
Updated the comments and rebased the branch with master. Thanks.

**Pull request merged**